### PR TITLE
added i18n

### PR DESCRIPTION
**Pull request by norbertschuler**
This is a just a simple test of https://react.i18next.com/ and https://www.i18next.com/ integration (mostly copied from https://lazertechnologies.com/blog/how-to-add-localization-to-your-nextjs-app).

Read the manual to learn more about:
- https://react.i18next.com/guides/multiple-translation-files
- https://react.i18next.com/guides/extracting-translations
- https://www.i18next.com/translation-function/interpolation
- https://www.i18next.com/translation-function/formatting
- https://www.i18next.com/plurals.html

Open issues:
- [ ] https://github.com/Plant-for-the-Planet/planet-webapp/pull/236#issuecomment-697180410
- [ ] https://github.com/Plant-for-the-Planet/planet-webapp/pull/236#issuecomment-698202800
- [ ] https://github.com/Plant-for-the-Planet/planet-webapp/pull/236#issuecomment-698203271
- [x] https://github.com/Plant-for-the-Planet/planet-webapp/pull/236#issuecomment-698150755
- [x] https://github.com/Plant-for-the-Planet/planet-webapp/pull/236#issuecomment-698726776

### CODE REVIEW

File: pages/_app.tsx [hunk 1]
```diff
@@ -41,10 +48,14 @@ export default function PlanetWeb({ Component, pageProps }: any) {
...
   }, []);
 
-  return (
-    <ThemeProvider>
-      <CssBaseline />
-      <Component {...pageProps} />
-    </ThemeProvider>
-  );
+  if (!initialized) {
+    return <p>Loading...</p>;
```
Comment (norbertschuler) [inline pages/_app.tsx:52]: I know this looks ugly!

File: pages/me.tsx [hunk 2]
```diff
@@ -1,16 +1,18 @@
...
-import { getMe } from '../public/locales/getTranslations';
 import Layout from '../src/features/common/Layout';
 import UserPage from '../src/features/user/UserProfile';
+import i18next from '../i18n';
+
+const { Trans, useTranslation } = i18next;
 
 export default function UserProfile() {
   const [userprofile, setUserprofile] = React.useState({});
+  const { t } = useTranslation(['me', 'common']);
```
Comment (norbertschuler) [inline pages/me.tsx:10]: Here you can see how to use multiple json files for translations.

File: pages/me.tsx [hunk 3]
```diff
@@ -79,8 +81,8 @@ export default function UserProfile() {
   return (
     <Layout>
       <h2 style={{ marginTop: '80px' }}>
-        description:
-        {texts.description}
+        description: {t('me:description', { name: 'Norbert' })}<br/>
```
Comment (norbertschuler) [inline pages/me.tsx:84]: Here you can see how to use interpolation.

**Comment by Plant-for-the-Planet** [comment]
- [ ] changing language does not work ; Current  language should be identified using browser, or using  country-map 
German is default language for --> DE,AT,CH  (Germany, Austria, Switzerland)

Language can be stored in localStorage.

**Comment by norbertschuler** [comment]
> Keep:` Seeding your Donation`

I have not understood this phrase like putting seeds into the ground? Therefore I changed it to "processing".

**Comment by Plant-for-the-Planet** [comment]
EUR 1.00 should be localized 

1,00

Screen Shot 2020-09-23 at 9 13 54 AM

File: public/static/locales/de/donate.json [hunk 4]
```diff
@@ -55,5 +55,5 @@
   "plantTreesAtURL": "Pflanze Bäume unter {{url}}",
   "copiedToClipboard": "Text in die Zwischenablage kopiert!",
   "noProjectsFound": "Keine Projekte gefunden",
-  "inTimeOfTaxReturns": ""
+  "inTimeOfTaxReturns": "bekommen Sie rechtzeitig für ihre Steuererklärung"
```
Comment (norbertschuler) [inline public/static/locales/de/donate.json:58]: I made all the translations with "Du" and not "Sie" in the German language. We should keep that style and do not mix it!

Comment (norbertschuler) [reply public/static/locales/de/donate.json:58]: Besides we really should avoid fragments of sentences as some languages want to change the order of the words or the translators can not understand the context.

Comment (sarveshpro) [reply public/static/locales/de/donate.json:58]: Can you please make that change in Lingohub or let me know the string I'll update.

Comment (norbertschuler) [reply public/static/locales/de/donate.json:58]: Changed it in LingoHub.

Comment (Plant-for-the-Planet) [reply public/static/locales/de/donate.json:58]: we go with Du. I've also asked Felix to review Lingohub

**Comment by norbertschuler** [comment]
- [ ] Missing translation for "PROJECTS" at https://github.com/Plant-for-the-Planet/planet-webapp/blob/6e4864a2047d036de7b6bb7e55f98c795a877604/src/features/public/PublicUserProfile/components/ProjectsContainer.tsx#L24

**Comment by sagararyal** [comment]
Translation is missing for read more

**Comment by norbertschuler** [comment]
- [ ] The HTML language attribute at https://github.com/Plant-for-the-Planet/planet-webapp/blob/6e4864a2047d036de7b6bb7e55f98c795a877604/pages/_document.tsx#L15 is not localized - there is a discussion about doing this at https://github.com/isaachinman/next-i18next/issues/20 but that does not seem to work for static pages. Maybe someone know a good solution, also for changing this value when the user changes the languages.

**Comment by norbertschuler** [comment]
- [ ] Also this meta-data locale value is not yet localized: https://github.com/Plant-for-the-Planet/planet-webapp/blob/6e4864a2047d036de7b6bb7e55f98c795a877604/pages/t/%5Bid%5D.tsx#L70

**Comment by harshvitra** [comment]
> * [ ]  The HTML language attribute at https://github.com/Plant-for-the-Planet/planet-webapp/blob/6e4864a2047d036de7b6bb7e55f98c795a877604/pages/_document.tsx#L15
>    is not localized - there is a discussion about doing this at [isaachinman/next-i18next#20](https://github.com/isaachinman/next-i18next/issues/20) but that does not seem to work for static pages. Maybe someone know a good solution, also for changing this value when the user changes the languages.

Don't have the solution but anyone working should know that this is not the solution - 

```
state = {
    language: 'en',
  }

  componentDidMount() {
    if (localStorage.getItem('i18nextLng')) {
      this.setState({language: localStorage.getItem('i18nextLng')});
    }
  }
```

**Comment by norbertschuler** [comment]
- [ ] Translation is missing for https://github.com/Plant-for-the-Planet/planet-webapp/blob/develop/src/features/public/Donations/components/ShareOptions.jsx

**Comment by sarveshpro** [comment]
> > * [ ]   The HTML language attribute at https://github.com/Plant-for-the-Planet/planet-webapp/blob/6e4864a2047d036de7b6bb7e55f98c795a877604/pages/_document.tsx#L15
> >   
> >   is not localized - there is a discussion about doing this at [isaachinman/next-i18next#20](https://github.com/isaachinman/next-i18next/issues/20) but that does not seem to work for static pages. Maybe someone know a good solution, also for changing this value when the user changes the languages.
> 
> Don't have the solution but anyone working should know that this is not the solution -
> 
> ```
> state = {
>     language: 'en',
>   }
> 
>   componentDidMount() {
>     if (localStorage.getItem('i18nextLng')) {
>       this.setState({language: localStorage.getItem('i18nextLng')});
>     }
>   }
> ```

@harshvitra Did you check this issue, it has a couple of solutions we can use.
https://github.com/isaachinman/next-i18next/issues/20

**Comment by sarveshpro** [comment]
> * [ ]  Also this meta-data locale value is not yet localized: https://github.com/Plant-for-the-Planet/planet-webapp/blob/6e4864a2047d036de7b6bb7e55f98c795a877604/pages/t/%5Bid%5D.tsx#L70

`og:locale:alternate` can be used to set other translated languages and the default language can be defined in tenant config. 
https://ogp.me/#optional:~:text=og%3Alocale%20%2D%20The%20locale%20these%20tags,locales%20this%20page%20is%20available%20in.

**Comment by norbertschuler** [comment]
> isaachinman/next-i18next#20

I tried some of that stuff, but none of them worked for me. I thought that's because we use static pages. Please feel free to test it yourself. If you found something working, please let's use it.

**Review by harshvitra — APPROVED** [review]

**Pull request merged**